### PR TITLE
feat: add ability to reject connection with ECONNREFUSED

### PR DESCRIPTION
**Pull request by viglucci**
Addresses #72

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -108,11 +108,22 @@ Mitm.prototype.connect = function connect(orig, Socket, opts, done) {
...
-  setTimeout(server.emit.bind(server, "connect"))
+  if (client.rejected) {
+    const connectionRefusedError = new Error()
+    connectionRefusedError.address = "127.0.0.1"
+    connectionRefusedError.code = "ECONNREFUSED"
+    connectionRefusedError.errno = -4078
+    connectionRefusedError.port = 9090
+    connectionRefusedError.syscall = "connect"
+    setTimeout(client.emit.bind(client, "error", connectionRefusedError))
+    setTimeout(client.emit.bind(client, "close", true))
```
Comment (viglucci) [inline index.js:119]: Node docs say that `close` should be emitted immediately after `error`.

**Comment by vkruoso** [comment]
This would be really great to test edge cases. This happens in real life and it is hard to test it.
